providers/base/bin/camera_quality_test.py

```diff
@@ -22,22 +22,51 @@
 import argparse
 from collections import deque
 import logging
+import os
 import time
 import sys
 
 import cv2
 import numpy as np
 
 from checkbox_support.vendor.brisque.brisque import BRISQUE
+from checkbox_support.helpers.timeout import timeout
 from tempfile import NamedTemporaryFile
 
+
 THRESHOLD = 60
 TIMEOUT = 10
 MIN_INTERVAL = 0.5
+PLAINBOX_SESSION_SHARE = os.environ.get("PLAINBOX_SESSION_SHARE", "")
 
 logger = logging.getLogger("camera_quality_test")
 
 
+def save_image(image: np.ndarray, device: str):
+    """Saves the image to the PLAINBOX_SESSION_SHARE directory if is set,
+    otherwise it saves the image to a temporary file.
+
+    Args:
+        image (np.ndarray): image to save
+        device (str): name of the video device
+
+    """
+    if PLAINBOX_SESSION_SHARE:
+        filepath = os.path.join(
+            PLAINBOX_SESSION_SHARE, "quality_image_{}.jpg".format(device)
+        )
+        cv2.imwrite(filepath, image)
+        print("Saved image to {}".format(filepath))
+    else:
+        with NamedTemporaryFile(
+            prefix="quality_image_{}".format(device),
+            suffix=".jpg",
+            delete=False,
+        ) as f:
+            cv2.imwrite(f.name, image)
+            print("Saved image to {}".format(f.name))
```

Comment:
If you use a `--output/-o` argument, I don't think the `else` logic is very useful. Moreover, the logic to check for the output folder could be done outside of the `save_image()` function.